ksqldb-common/src/test/java/io/confluent/ksql/function/UdfIndexTest.java

```diff
@@ -964,6 +936,60 @@ public void shouldThrowIfNoExactMatchAndImplicitCastDisabled() {
             + "(INTEGER)"));
   }
 
+  @Test
+  public void shouldThrowWhenGivenVagueImplicitCast() {
```

Comment:
Minor suggestion (slightly more concise):
```suggestion
  public void shouldThrowOnVagueImplicitCast() {
```

Reply:
Though I see that there's another negative test case for a different ambiguous implicit cast below. We could rename this one to `shouldThrowOnAmbiguousImplicitCastWithoutGenerics()` to contrast with the other one?